src/controller/CHIPDevice.h

```diff
@@ -89,11 +90,13 @@ class DLL_EXPORT Device
      * @brief
      *   Send the provided message to the device
      *
-     * @param[in] message   The message to be sent.
+     * @param[in] protocolId  The protocol identifier of the CHIP message to be sent.
+     * @param[in] msgType     The message type of the corresponding protocol.
+     * @param[in] message     The message to be sent.
      *
      * @return CHIP_ERROR   CHIP_NO_ERROR on success, or corresponding error
      */
-    CHIP_ERROR SendMessage(System::PacketBufferHandle message);
+    CHIP_ERROR SendMessage(Protocols::Id protocolId, uint8_t msgType, System::PacketBufferHandle message);
```

Comment:
We should add a template helper that infers the protocol id from a strongly typed message type....  and cosider making the untyped method protected.  Followup is fine for this.

Reply:
https://github.com/project-chip/connectedhomeip/issues/5672